README.md

```diff
@@ -2,58 +2,146 @@
+      namespace: <your-cluster-id>
+  version: 0.6.0
+
+```
+
+When installing via the Giant Swarm web UI, you'll need to apply the above labels and annotations manually.
+
+### Step 4: After deployment
+
+- Disable proxy injections for pods running in the `kube-system` namespace by applying certain labels and annotations
+
+```bash
+kubectl annotate namespace kube-system linkerd.io/inject=disabled
+kubectl label namespace kube-system config.linkerd.io/admission-webhooks=disabled
+```
+
+- Optional but recommended: You can use the `linkerd` cli as usual with this app as we're using the default namespaces. (`linkerd` and `linkerd-cni`). You can download it from the [linkerd release page](https://github.com/linkerd/linkerd2/releases/tag/stable-2.10.2).
+
+- We strongly recommend installing the `linkerd viz` extension using the [`linkerd`](#usage-with-linkerd-cli) by executing
+
+```bash
+linkerd viz install | kubectl apply -f -
+```
+
+After installation, you can open the dashboard by executing
+
+```
+linkerd viz dashboard
+```
+
+## Mesh your workloads
+
+After installation, linkerd looks for a `linkerd.io/inject: enabled` annotation on `Namespaces` or other workload resources. Adding this annotation to your workload namespaces will trigger automatic proxy container injection to your pods. More information on proxy injection can be found on the ["Automatic Proxy Injection" page](https://linkerd.io/2.10/features/proxy-injection/) in the upstream documentation.
+
+**Attention**: Proxy containers are using `EmptyDir` volumes for storing ephemeral data, so all of your workload pods meshed by linkerd require a `PodSecurityPolicy` which allows use of `EmptyDir` volumes.
+
+## Installing without the CNI plugin
+
+In order to install this app without the CNI plugin, you'll need to specify `cniEnabled: false` in your user configuration.
+
+Be aware that running without the CNI plugin, proxy containers will run as `root` and will require `NET_ADMIN` and `NET_RAW` capabilities.
+
+## Troubleshooting
+
+### App installation fails when I specify a namespace name other than `linkerd`
 
-If you are deploying this chart on top of Linkerd's CNI driver then you must enable this
-behaviour by setting `global.cniEnabled: true`.
+The `linkerd` namespace name is assumed in several parts of the chart, so we restricted installation to namespaces named `linkerd` only.
 
-### Note:
+### Proxy containers fail to start
 
-`values.yaml` uses [YAML anchors](https://helm.sh/docs/chart_template_guide/yaml_techniques/#yaml-anchors) to
-be DRY where possible, however this can cause problems when these values are overridden by a custom
-values file. If you wish to enabled or disable deployment of the `grafana` and `tracing` subcharts
-then you should override the values at `.$subchart.enabled`, **not** `.global.$subchart.enabled`.
+Your workload must be able to create `EmptyDir` volumes, so you'll need to create a `PodSecurityPolicy` allowing the creation of `EmptyDir` volumes.
```

Comment:
We should add info here about how to edit the default PSP to easily allow for `emptyDirs` everywhere. Doing this one-by-one for every deployment is cumbersome.

Reply:
added :heavy_check_mark: